internal/configs/hcl2shim/mock_value_composer.go

```diff
@@ -174,8 +177,8 @@ func (mvc MockValueComposer) composeMockValueForBlocks(schema *configschema.Bloc
 		if hasDefaultVal && !defaultVal.Type().IsObjectType() {
 			hasDefaultVal = false
 			diags = diags.Append(tfdiags.WholeContainingBody(
-				tfdiags.Warning,
-				fmt.Sprintf("Ignored mock/override field `%v`", k),
+				tfdiags.Error,
+				fmt.Sprintf("Invalid override for block field `%v`", k),
 				fmt.Sprintf("Blocks can be overridden only by objects, got `%s`", defaultVal.Type().FriendlyName()),
 			))
 		}
```

Comment:
Since this is an error now, we don't want to proceed so let's put a `continue` here. We can also remove `hasDefaultVal = false`.